### ISSUE TITLE
Nullability check incorrectly on `new` expression with a condition

### ISSUE THREAD

**Issue by SunnieShine** [dotnet/roslyn#49118]
**Version Used**: 
VS 16.8 Preview 6

**Steps to Reproduce**:

Type code

```csharp
#nullable enable

bool condition = true; // May be some other condition.

// ...
// Some code that doesn't modify 'condition'.
// ...

ClassObject? co = condition ? new() : null;
if (condition) // Here allows more conditions using logical and operator '&&'.
{
    co.SomeMethod(); // CS8602.
}

public class ClassObject
{
    public void SomeMethod() { }
}
```

[This link](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBMQGoA+BiAdgVwBtCJhC4ACOXU8gWAChHgB7FwigYxdxQEsYfHhQC8FGAnxwA3BQD0cigFkIATwrBKUFmEosYAC0Rce/QTwB0jRgooX7NxQGUdlbikqGIMCihZwoXAByHzAWfgAzdSDuXgEhYKsGW3skxgBhEigoAHlgACs4ThgAfhNREzjzXAoy3DgAdwAKAEoKEAoCYmlGPgiKJtizBLbbAAlESghiFgaoCjCEN1N4nnmAAQAGCkIWAHM+TmmKCF4KFgAHRG8WBAoggDIHoKSAb0YKT5MLF10lOEM4Vasls6ScAA4AGybABMSQAvtYGOsAMwUdYwiiZaC5ApFGCMd4ML7otHrAAsFF+cH+gJQrQorwoiIY8KAA==)

**Expected Behavior**:

`co` is currently not `null`, so CS8602 shouldn't be raised.

**Actual Behavior**:

CS8602 raised.

**Comment by Joe4evr** [comment]
Nullable analysis does not keep track of flags like this. See also #48553

**Comment by SunnieShine** [comment]
@Joe4evr Okay. Thank you for your reply.
